src/aics-image-viewer/components/ViewerStateProvider/index.tsx

```diff
@@ -159,23 +161,22 @@ const ViewerStateProvider: React.FC<{ viewerSettings?: Partial<ViewerState> }> =
     setChannelSettings(newChannelSettings);
   }, []);
 
-  // Sync viewer settings prop with state
-  // React docs seem to be fine with syncing state with props directly in the render function:
-  // https://react.dev/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes
-  if (props.viewerSettings) {
-    let newSettings = viewerSettings;
+  useEffect(() => {
+    if (props.viewerSettings) {
+      let newSettings = viewerSettings;
 
-    for (const key of Object.keys(props.viewerSettings) as (keyof ViewerState)[]) {
-      if (viewerSettings[key] !== props.viewerSettings[key]) {
-        // Update viewer settings one at a time to allow change handlers to keep state valid
-        newSettings = applyChangeToViewerSettings(viewerSettings, key, props.viewerSettings[key] as any);
+      for (const key of Object.keys(props.viewerSettings) as (keyof ViewerState)[]) {
+        if (viewerSettings[key] !== props.viewerSettings[key]) {
+          // Update viewer settings one at a time to allow change handlers to keep state valid
```

Comment:
I know you didn't write this but why not `newSettings[key] !== props.viewerSettings[key]`? just so it's clear which one is the new one?

Reply:
Makes sense to me 👍